dbt_subprojects/dex/macros/models/_project/zeroex/zeroex_v2.sql

```diff
@@ -341,6 +346,7 @@ select  block_time,
         settler_address as contract_address 
     from maker_logs
     join zeroex_tx st using (block_time, block_number, tx_hash, rn, settler_address) 
+    left join {{ source( 'evms', 'contracts') }} c on c.address = st.taker and c.blockchain = blockchain
```

Comment:
rather than evms schema view here, can you point to chain specific versions instead? you can use blockchain param from models to supply to schema name

Reply:
I made this change and remembered why I used evms here, after running into an error again. 

There's no table 'mode.contracts' so it throws an error on that chain. This join isn't required, but helps replace contract_addresses that may have been picked up instead of taker address. 

What you do you recommend here?

Reply:
you can wrap the join in jinja if condition for blockchain param.
if blockchain = mode, then ignore, else do join

there are some examples of this in spellbook if you search, such as gas fees macro

Reply:
thanks, all set now @jeff-dude